src/Oscoin/Storage/Receipt.hs

```diff
@@ -1,80 +1,67 @@
--- | Provides helper functions to easily implement 'MonadReceiptStore'
--- instances for State monads and IO Reader monads with handles.
+-- | This module provides the 'ReceiptStore' interface which is a
+-- 'ContentStore' that uses the transaction hashes to identify
+-- transaction receipts.
+--
+-- For legacy reasons we also provide the 'MonadReceiptStore' type
+-- class to obtain a receipt store.
 module Oscoin.Storage.Receipt
     ( MonadReceiptStore(..)
-    , Receipt(..)
-
-    -- * State based receipt store
-    , Store
-    , HasStore(..)
-    , emptyStore
-    , addWithStore
-    , lookupWithStore
-
-    -- * TVar handle based receipt store
-    , Handle
-    , HasHandle(..)
-    , newHandle
-    , addWithHandle
-    , lookupWithHandle
+    , addReceipt
+    , lookupReceipt
+
+    , ReceiptStore
+    , ReceiptStoreMap
+    , newReceiptStoreIO
+    , mkStateReceiptStore
+
+    -- * Re-exports from "Oscoin.Storage.ContentStore"
+    , storeContent
+    , lookupContent
+    , hoistContentStore
     ) where
 
 import           Oscoin.Prelude
 
+import           Lens.Micro
 import           Oscoin.Crypto.Blockchain.Eval (Receipt(..))
-import           Oscoin.Crypto.Hash (Hash)
 import qualified Oscoin.Crypto.Hash as Crypto
-import           Oscoin.Storage.Receipt.Class
-
-import           Control.Concurrent.STM
-import qualified Data.Map as Map
-import           Lens.Micro
-import           Lens.Micro.Mtl
-
--- State based MonadReceiptStore --------------------------------------
-
-type Store c tx o = Map.Map (Crypto.Hashed c tx) (Receipt c tx o)
+import           Oscoin.Storage.ContentStore
 
-class HasStore c tx o r where
-    storeL :: Lens' r (Store c tx o)
+-- | 'ContentStore' for 'Receipt's where a receipt is adressed by the
+-- hash of the transaction the receipt is for.
+type ReceiptStore c tx o m = ContentStore (Crypto.Hashed c tx) (Receipt c tx o) m
 
-emptyStore :: Store c tx o
-emptyStore = Map.empty
+-- | Underlying data structure for pure versions of 'ReceiptStore'
+-- created with 'mkStateReceiptStore'.
+type ReceiptStoreMap c tx o = Map (Crypto.Hashed c tx) (Receipt c tx o)
```

Comment:
Minor: reading the documentation makes clear what this is about, but I guess this might be confusing when encountered "in the wild" and might confuse the reader trying to understand the different between a `ReceiptStore` and a `ReceiptStoreMap`. Do you think moving this into a separate `Pure` module might makes sense, or name it like `PureContentStore` or something like that?

Follow up (as I keep writing something and scrapping it 😅 ):if I understand correctly, this is exported and used _only_ in tests (orthogonal thought: we should really tame those tests so that we don't need to introduce these ad-doc types). If that's true, methinks it's better if we name this _explicitly_ (as in `TestReceiptStore`) to convey the fact this is something which needs to be used _only_ for testing?

Thoughts?

Reply:
What do you think about calling this `ReceiptMap`. I could also remove it (or not export it).

Reply:
I am a firm believer of code which states intent, so my preference would still be to name it "TestBlah", or to move it into the test code (if that works) while keeping the name as-it-is, if you like it better 😁 

But I am not forcing refactorings on people, so definitely feel free to act according to your best judgement (and don't feel this should stop you from merging this PR altogether).

Reply:
I had the same thought while reading. `ReceiptMap` would make more sense, or as @adinapoli-mndc proposed, moving it to a `Pure` module.

Reply:
I’m going to rename it to `ReceiptMap`. I don’t think moving it into a `Pure` module would make sense since that module would only contain `ReceiptMap` and `mkStateReceiptStore`.